packages/blade/src/components/BottomSheet/BottomSheet.web.tsx

```diff
@@ -82,6 +83,8 @@ const _BottomSheet = ({
   const [headerHeight, setHeaderHeight] = React.useState(0);
   const [footerHeight, setFooterHeight] = React.useState(0);
   const [grabHandleHeight, setGrabHandleHeight] = React.useState(0);
+  const [hasBodyPadding, setHasBodyPadding] = React.useState(true);
+  const [isHeaderEmpty, setIsHeaderEmpty] = React.useState(false);
```

Comment:
BottomSheet.web and BottomSheet.native seems to share a lot of state code. Should we move this out into a custom hook?

Reply:
On the surface, yes - it does share few bits of state code and logic. 

But in practice how we are using these states and the logic between native & web implementation has subtle differences which if we extract out will cause more overhead and introduce `ifocalypse`. 

Discussed this earlier with chaitanya also, it's better to keep things separated out for now looking at the complexity of the two platforms.